### PR TITLE
Create CredscanProcess.md

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: doc/dev/CredscanProcess.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+2. Each warning will begin with the path to the file containing a potential credential, as well as the row and column where
+the credential string begins. For example, for a potential credential that starts in row 3 and column 20 of a
+particular file:
+```
+##[error]sdk/{service}/{package}/{file}.json:sdk/{service}/{package}/{file}.json(3,20)
+```
+
+The warning will then list an error code and description of why the potential credential was flagged.
+
+## True Positives
```
Comment (sima-zhu) [inline doc/dev/CredscanProcess.md:28]: We have a general guidance on how we mitigate the issue. I have a link in wiki: aka.ms/credscan. Better to include the information there, as we need to rotate the secret besides of sanitizer.

File: doc/dev/CredscanProcess.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+3. Add the fields that need to be sanitized into the [RPName]RecordedTestSanitizer.cs file into the constructor as shown [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/CosmosDBManagementRecordedTestSanitizer.cs#L14).
+3.	If the protected constructor in your [RPName]ClientBase class does not create a [RPName]RecordedTestSanitizer object, add the initialization [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/CosmosDBManagementClientBase.cs#L48).
+
+Once you have added the Sanitizers, re-record all your test files that had leaked secrets. They should be replaced with the keyword “Sanitized”.
+## False Positives
```
Comment (sima-zhu) [inline doc/dev/CredscanProcess.md:43]: I think the sanitizer approach also applies some false positive cases.

File: doc/dev/CredscanProcess.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+Ideally, fake credential files -- which contain nothing but fake secrets -- should be suppressed and their fake
+credentials shouldn't appear in any other files. Sanitizers should be used to keep fake credentials out of test
+recordings when possible. String value suppression should be avoided unless the string appears in many files.
+
+Suppressing string values will disable warnings no matter where the string comes up during a scan, but is inefficient
+and inconvenient for lengthy strings. Suppressing warnings in a file is convenient for fake credential files, but
+strings in that file will still trigger warnings if present in another unsuppressed file.
+
+## Support
+If you have any questions about CredScan in azure-sdk-for-net, please post your question in this [Teams channel](https://teams.microsoft.com/l/channel/19%3a7b87fb348f224b37b6206fa9d89a105b%40thread.skype/Language%2520-%2520DotNet?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47).
```
[the remark this answers is not in the document]
Comment (nisha-bhatia) [reply doc/dev/CredscanProcess.md]: Moved to internal

File: doc/dev/CredscanProcess.md (hunk 2 above)
Comment (nisha-bhatia) [reply doc/dev/CredscanProcess.md:32]: Moved to internal

File: doc/dev/CredscanProcess.md (hunk 1 above)
Comment (nisha-bhatia) [reply doc/dev/CredscanProcess.md:28]: I think the aka.ms link is in the first paragraph so hopefully that is ok, moving other information to internal

Comment (sima-zhu) [reply doc/dev/CredscanProcess.md:28]: aka.ms/credscan did not say anything about rotate the secret.

This is the link talking about this: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/101/Leaked-secret-procedure

**Review by sima-zhu — APPROVED** [review]
LGTM

File: doc/dev/CredscanProcess.md (hunk 1 above)
Comment (nisha-bhatia) [reply doc/dev/CredscanProcess.md:28]: I copied over the leaked secret procedure into this doc (I wasn't sure if I could link to Internal link) so hopefully it's ok now.

**Pull request merged**